src/pages/ProfilePage.tsx

```diff
@@ -21,154 +20,136 @@ import PressableWithoutFocus from '@components/Pressable/PressableWithoutFocus';
 
-const matchType = PropTypes.shape({
-    params: PropTypes.shape({
-        /** accountID passed via route /a/:accountID */
-        accountID: PropTypes.string,
+type ProfilePageOnyxProps = {
+    /** The personal details of the person who is logged in */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
 
-        /** report ID passed */
-        reportID: PropTypes.string,
-    }),
-});
-
-const propTypes = {
-    /* Onyx Props */
-
-    /** The personal details of all users */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
-    /** Route params */
-    route: matchType.isRequired,
-
-    /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user accountID */
-        accountID: PropTypes.number,
-    }),
-
-    ...withLocalizePropTypes,
+    report: OnyxEntry<Report>;
 };
 
-const defaultProps = {
-    // When opening someone else's profile (via deep link) before login, this is empty
-    personalDetails: {},
-    session: {
-        accountID: 0,
-    },
-};
+type ProfilePageProps = ProfilePageOnyxProps &
+    StackScreenProps<ProfileNavigatorParamList, typeof SCREENS.PROFILE_ROOT> & {
+        /** Session info for the currently logged in user. */
+        session: OnyxEntry<Session>;
+    };
 
 /**
  * Gets the phone number to display for SMS logins
- *
- * @param {Object} details
- * @param {String} details.login
- * @param {String} details.displayName
- * @returns {String}
  */
-const getPhoneNumber = (details) => {
-    // If the user hasn't set a displayName, it is set to their phone number, so use that
-    const displayName = lodashGet(details, 'displayName', '');
+const getPhoneNumber = ({login = '', displayName = ''}: PersonalDetails): string | undefined => {
+    // If the user hasn't set a displayName, it is set to their phone number
+
     const parsedPhoneNumber = parsePhoneNumber(displayName);
     if (parsedPhoneNumber.possible) {
-        return parsedPhoneNumber.number.e164;
+        return parsedPhoneNumber?.number?.e164;
     }
 
     // If the user has set a displayName, get the phone number from the SMS login
-    return details.login ? Str.removeSMSDomain(details.login) : '';
+    return login ? Str.removeSMSDomain(login) : '';
 };
 
-function ProfilePage(props) {
+function ProfilePage({personalDetails = {}, route, session, report}: ProfilePageProps) {
     const styles = useThemeStyles();
-    const accountID = Number(lodashGet(props.route.params, 'accountID', 0));
-    const details = lodashGet(props.personalDetails, accountID, ValidationUtils.isValidAccountRoute(accountID) ? {} : {isloading: false});
+    const {translate, formatPhoneNumber} = useLocalize();
+    const accountID = Number(route.params?.accountID ?? 0);
+    const details: PersonalDetails =
+        personalDetails?.[accountID] ?? (ValidationUtils.isValidAccountRoute(accountID) ? {accountID: 0, avatar: ''} : {isLoading: false, accountID: 0, avatar: ''});
```

Comment:
Are you sure this update doesn't have any side affects? Maybe it will be more save to have it like
```suggestion
    const details: PersonalDetails | EmptyObject =
        personalDetails?.[accountID] ?? (ValidationUtils.isValidAccountRoute(accountID) ? {} : {isLoading: false, accountID: 0, avatar: ''});
```